crates/router/src/core/poll.rs

```diff
@@ -0,0 +1,50 @@
+use api_models::poll::{PollResponse, PollStatus};
+use common_utils::ext_traits::StringExt;
+use error_stack::ResultExt;
+use router_env::{instrument, tracing};
+
+use super::errors;
+use crate::{
+    core::errors::RouterResponse,
+    services::{logger, ApplicationResponse},
+    AppState,
+};
+
+#[instrument(skip_all)]
+pub async fn retrieve_poll_status(
+    state: AppState,
+    req: crate::types::api::PollId,
+) -> RouterResponse<PollResponse> {
+    let redis_conn = state
+        .store
+        .get_redis_conn()
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to get redis connection")?;
+    let request_poll_id = req.poll_id;
+    // prepend 'poll_' to restrict access to only fetching Poll IDs, as this is a freely passed string in the request
+    let poll_id = format!("poll_{}", request_poll_id);
```

Comment:
how are we restricting access control for poll ids? A merchant with a publishable key can access poll status of other merchants?